plugins/outputs/whatap/whatap.go

```diff
@@ -0,0 +1,382 @@
+	crc = crc ^ 0xffffffff
+	return int32(crc)
+}
+func Hash64Str(str string) int64 {
+	return Hash64Bytes([]byte(str))
+}
+func Hash64Bytes(bytes []byte) int64 {
+	crc := uint64(0xffffffffffffffff)
+	sz := len(bytes)
+	for i := 0; i < sz; i++ {
+		b := bytes[i]
+		crc = (crc)>>8 ^ uint64(int32(table[uint8(crc^uint64(b))]))
+	}
+	crc = crc ^ 0xffffffffffffffff
+	return int64(crc)
+}
+
+type TagCountPack struct {
+	Pcode    int64
+	Oid      int32
+	Time     int64
+	Category string
+	tagHash  int64
+	Tags     map[string]string
+	Data     map[string]interface{}
+}
+
+func NewTagCountPack() *TagCountPack {
+	p := new(TagCountPack)
+	p.Tags = make(map[string]string)
+	p.Data = make(map[string]interface{})
+	return p
+}
+
+func (this *TagCountPack) GetPackType() int16 {
+	return PACK_TAG_COUNT
+}
+func (this *TagCountPack) PutTag(name, val string) {
+	this.Tags[name] = val
+}
+func (this *TagCountPack) Put(name string, v interface{}) {
+	this.Data[name] = v
+}
+func (this *TagCountPack) Write(dout *DataOutputX) {
+	dout.WriteDecimal(this.Pcode)
+	dout.WriteInt(this.Oid)
+	dout.WriteLong(this.Time)
+	dout.WriteByte1(0)
+	dout.WriteText(this.Category)
+
+	// Outputs the tag information of the matrix to a byte array
+	// and converts it to an int64 hash.
+	// The server caches or indexes data with that hash value.
+	tagIO := NewDataOutputX()
+	this.WriteTag(tagIO)
+	tagBytes := tagIO.ToByteArray()
+	this.tagHash = Hash64Bytes(tagBytes)
+
+	dout.WriteDecimal(this.tagHash)
+	dout.WriteBytes(tagBytes)
+
+	this.WriteData(dout)
+}
+
+func (this *TagCountPack) WriteTag(dout *DataOutputX) {
+	//sort
+	keys := make([]string, 0)
+	for k := range this.Tags {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	dout.WriteByte1(VALUE_MAP)
+	dout.WriteDecimal(int64(len(this.Tags)))
+	for _, k := range keys {
+		dout.WriteText(k)
+		// writeValue - Text
+		dout.WriteByte1(VALUE_TEXT)
+		dout.WriteText(this.Tags[k])
+	}
+}
+
+func (this *TagCountPack) WriteData(dout *DataOutputX) {
+	//sort
+	keys := make([]string, 0)
+	for k := range this.Data {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	dout.WriteByte1(VALUE_MAP)
+	dout.WriteDecimal(int64(len(this.Data)))
+	for _, k := range keys {
+		dout.WriteText(k)
+		// writeValue - Text
+		v := this.Data[k]
+		switch v.(type) {
+		case int, int16, int32, int64:
+			dout.WriteByte1(VALUE_DECIMAL)
+			dout.WriteDecimal(v.(int64))
+		case uint, uint16, uint32, uint64:
+			dout.WriteByte1(VALUE_DECIMAL)
+			dout.WriteDecimal(int64(v.(uint64)))
+		case float32, float64:
+			dout.WriteByte1(VALUE_DOUBLE)
+			dout.WriteDouble(v.(float64))
+		case string:
+			dout.WriteByte1(VALUE_TEXT)
+			dout.WriteText(v.(string))
+		default:
+		}
+	}
+}
+
+func newWhatap() *Whatap {
+	hostname, _ := os.Hostname()
+	return &Whatap{
+		Timeout: 60 * time.Second,
+		Session: TcpSession{},
+		Oname:   hostname,
+		Oid:     HashStr(hostname),
+	}
+}
+func init() {
+	outputs.Add("whatap", func() telegraf.Output {
+		return newWhatap()
+	})
+
+}
```

Comment:
Please merge the two
```suggestion
}
func init() {
	hostname, _ := os.Hostname()
	outputs.Add("whatap", func() telegraf.Output {
		return &Whatap{
			Timeout: 60 * time.Second,
			Session: TcpSession{},
			Oname:   hostname,
			Oid:     HashStr(hostname),
		}
	})

}
```

Reply:
Edited and applied. We ask for confirmation.